specification/hybridconnectivity/resource-manager/Microsoft.HybridConnectivity/PublicCloud/stable/2024-12-01/publicCloud.json

```diff
@@ -0,0 +1,1671 @@
+            "schema": {
+              "$ref": "#/definitions/SolutionTypeResourceListResult"
+            }
+          },
+          "default": {
+            "description": "An unexpected error response.",
+            "schema": {
+              "$ref": "../../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "SolutionTypes_ListByResourceGroup": {
+            "$ref": "./examples/SolutionTypes_ListByResourceGroup.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HybridConnectivity/solutionTypes/{solutionType}": {
+      "get": {
+        "operationId": "SolutionTypes_Get",
+        "tags": [
+          "SolutionTypes"
+        ],
+        "description": "Get a SolutionTypeResource",
+        "parameters": [
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "solutionType",
+            "in": "path",
+            "description": "Solution Type resource",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Azure operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/SolutionTypeResource"
+            }
+          },
+          "default": {
+            "description": "An unexpected error response.",
+            "schema": {
+              "$ref": "../../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "SolutionTypes_Get": {
+            "$ref": "./examples/SolutionTypes_Get.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "AwsCloudProfile": {
+      "type": "object",
+      "description": "cloud profile for AWS.",
+      "properties": {
+        "accountId": {
+          "type": "string",
+          "description": "Account id for the AWS account.",
+          "x-ms-mutability": [
+            "create"
+          ]
+        },
+        "excludedAccounts": {
+          "type": "array",
+          "description": "List of AWS accounts which need to be excluded.",
+          "items": {
+            "type": "string"
+          }
+        },
+        "isOrganizationalAccount": {
+          "type": "boolean",
```

Comment:
>boolean [](http://example.com/codeflow?start=19&length=7)

how about an enum of account type with values : Organizational and other .


Standard guidance is: replace boolean/switch properties with a more meaningful enum whenever possible. 

A boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated]. 

Note: do NOT define a 'boolean enum' with two values 'True and False'. This might be easier to 'extend' in terms of types, but semantically its cofusing, and no better than a boolean.

Reply:
We intentionally moved away from enums, since adding or removing enum values would require api-version change and that adds work to our GA planning

Reply:
Since this is the first time you are going GA, Its better to fix stuff and this likely  going to stay forever. How would you handle different Account types in the future other than organizational?

Reply:
We only have two types of accounts organizational and member accounts, so we went with boolean in this case